engine/engine.go

```diff
@@ -106,10 +112,14 @@ func (e *engine) NewInstantQuery(q storage.Queryable, opts *promql.QueryOpts, qs
 		return nil, err
 	}
 
+	if e.debugWriter != nil {
```

Comment:
NopWriter might be better to avoid nil checks.

Reply:
Hm, we don't want to spend time on explaining to nothing though, so check will be there. We can hide it later behind extra function if yo uwant